\documentclass[a4paper,9pt,DIV=14]{scrartcl}
% \documentclass[a4paper,10pt,DIV=14]{scrartcl}
% \documentclass[a4paper,10pt,DIV=13]{scrartcl} % scrreprt

\usepackage[utf8]{inputenc}
\setlength{\parskip}{0.8em}
\setlength{\parindent}{0pt}

% ######################################################################################################################
%         Custom Packages
% ######################################################################################################################

\usepackage{graphicx}
\usepackage[table]{xcolor}

\graphicspath{{../figures_pdf/}}

\usepackage[square,sort,comma,numbers]{natbib}
\usepackage[english]{babel}

% https://tex.stackexchange.com/questions/267675/pdftex-error-pdflatex-file-ecbx0800-font-ecbx0800-at-600-not-found
\usepackage{lmodern}

% https://reu.dimacs.rutgers.edu/Symbols.pdf
\usepackage{amssymb}

% https://tex.stackexchange.com/a/6105
\usepackage[binary-units=true]{siunitx}

% https://tex.stackexchange.com/questions/94845/problems-with-toprule-and-midrule-in-a-table
\usepackage{booktabs}

% https://ctan.org/pkg/relsize
\usepackage{relsize}

\usepackage[bf,format=plain]{caption}

% https://tex.stackexchange.com/questions/135358/changing-the-formatting-of-subcaption-for-reference
\usepackage[labelformat=simple]{subcaption}
\renewcommand\thesubfigure{(\alph{subfigure})}

% https://tex.stackexchange.com/a/39981
\usepackage[nolist,nohyperlinks]{acronym}

% http://bytesizebio.net/2013/03/11/adding-supplementary-tables-and-figures-in-latex/
\newcommand{\beginsupplement}{%
    \setcounter{table}{0}
    \renewcommand{\thetable}{S\arabic{table}}%
    \setcounter{figure}{0}
    \renewcommand{\thefigure}{S\arabic{figure}}%
}

\usepackage{amsmath}
\usepackage{xfrac}
\usepackage{bm}
\usepackage{scalerel,stackengine}

% Increase row spacing in align envs
% https://tex.stackexchange.com/a/14680/171851
\addtolength{\jot}{0.75em}

% Excluded the following lines from the pnas class file.
% Include them here again to ensure correct setup.
% See https://tex.stackexchange.com/q/1863
\usepackage[colorlinks=true, allcolors=blue]{hyperref}
\renewcommand\UrlFont{\color{black}\sffamily}

% https://tex.stackexchange.com/a/78020
\pdfsuppresswarningpagegroup=1

% \externaldocument[main:]{main}

% \usepackage{nameref,zref-xr}
% \zxrsetup{toltxlabel}
% \zexternaldocument*{pppp}

% https://tex.stackexchange.com/questions/180019/grouping-two-tables-one-above-the-other
\usepackage{hypcap} % fix the links

% https://tex.stackexchange.com/questions/109467/footnote-in-tabular-environment
\usepackage{threeparttable}
% \usepackage[flushleft]{threeparttable}

% LEFT-ALIGN THE AUTHOR LIST
% Overleaf syntax highlighting gets messed up by this, but it works.
% Commented out now to not mess with overleaf for now.
% https://tex.stackexchange.com/a/364432/171851
% \usepackage{xpatch}
% \makeatletter
% \xpatchcmd{\@maketitle}{\begin{center}}{\begin{flushleft}}{}{}
% \xpatchcmd{\@maketitle}{\end{center}}{\end{flushleft}}{}{}
% \xpatchcmd{\@maketitle}{\begin{tabular}[t]{c}}{\begin{tabular}[t]{@{}l@{}}}{}{}
% \makeatother

% ######################################################################################################################
%         Custom Commands
% ######################################################################################################################

% Preparation for the todo command
\newcounter{todo}
\setcounter{todo}{0}
\newcounter{popoolissue}
\setcounter{popoolissue}{0}

% ----------------------------------------------------------
% TODO command: 

% Use the first line below to show them,
% use the second to hide them,
% and comment out the respective other line.
% LaTeX is so easy ;-)

% \newcommand\todo[1]{{\stepcounter{todo}\color{purple}{TODO[\arabic{todo}]: #1}}}
\newcommand\todo[1]{}

% Same for "popoolissue", which we use to annotate our questions for the popoolation team
\newcommand\popoolissue[1]{{\stepcounter{popoolissue}\color{purple}{ITEM \arabic{popoolissue}: #1}}}
% \newcommand\popoolissue[1]{}
% ----------------------------------------------------------


% todo environment
% \newcounter{todo}
% \newcommand\todo[1]{\refstepcounter{todo}{\color{purple}{#1}}\addcontentsline{todolist}{subsection}{\thetodo:~#1}}
% un-comment the next line in order to hide all todos:
% \renewcommand\todo[1]{}
% \newcommand\moi[1]{{\color{orange}{#1}}}

% Suppress KOMA script warning
% https://tex.stackexchange.com/q/412368
\DeclareOldFontCommand{\bf}{\normalfont\bfseries}{\mathbf}

% https://nw360.blogspot.de/2007/12/rename-bibliography-title-in-latex.html
% https://tex.stackexchange.com/a/306268
% \renewcommand\refname{References}
% \renewcommand\bibname{References}
% \addto{\captionsenglish}{%
%   \renewcommand{\bibname}{References}
% }

% Define custom styles
\newcommand\toolname{\textsc}
\newcommand\taxonname{\textit}
\newcommand\formatname{\texttt}
\newcommand\langname{\texttt}

% \newcommand\figref[1]{Figure~\ref{#1}}
% \newcommand\tabref[1]{Table~\ref{#1}}
% \newcommand\eqnref[1]{Equation~(\ref{#1})}

\newcommand\figref[1]{Fig.~\ref{#1}}
\newcommand\tabref[1]{Tab.~\ref{#1}}
\newcommand\eqnref[1]{Eq.~(\ref{#1})}
\newcommand\secref[1]{Section~(\ref{#1})}

\newcommand{\thetapi}{\widehat{\theta_\pi}}
\newcommand{\watterson}{\widehat{\theta_\text{W}}}
\newcommand{\fst}{F\textsubscript{ST}}
\newcommand{\mathfst}{F_\text{ST}}
\newcommand{\neifst}{F_\text{ST}^\text{Nei}}
\newcommand{\hudsonfst}{F_\text{ST}^\text{Hudson}}
\newcommand{\neiestimator}{\widehat{F}_\text{ST}^\text{Nei}}
\newcommand{\hudsonestimator}{\widehat{F}_\text{ST}^\text{Hudson}}
\newcommand{\eg}{e.\,g.}
\newcommand{\ie}{i.\,e.}

\newcommand\citeay[1]{\citeauthor{#1} (\citeyear{#1}) \cite{#1}}

% https://tex.stackexchange.com/questions/103408/symbol-for-corresponds-to-equals-sign-with-hat
\newcommand\equalhat{%
\let\savearraystretch\arraystretch
\renewcommand\arraystretch{0.3}
\begin{array}{c}
\stretchto{
    \scalerel*[\widthof{=}]{\wedge}
    {\rule{1ex}{3ex}}%
}{0.5ex}\\ 
=%
\end{array}
\let\arraystretch\savearraystretch
}

% ######################################################################################################################
%         Title and Header
% ######################################################################################################################

% Title Page
\title{Supplementary Text: \\ Pool-Sequencing corrections for population genetic statistics}
% \subtitle{Genesis and Gappa: Library and Toolkit for \\Working with Phylogenetic (Placement) Data.}
\author{Lucas Czech, Jeffrey P. Spence, and Moisés Expósito-Alonso \\ Correspondence: \href{mailto:moisesexpositoalonso@gmail.com}{moisesexpositoalonso@gmail.com}, \href{mailto:luc@s-cze.ch}{luc@s-cze.ch}}
\date{}

% ######################################################################################################################
%         Document
% ######################################################################################################################

\begin{document}

% \begin{abstract}
% sflkdsjfkj sf
% \end{abstract}

\beginsupplement

% \maketitle

\begingroup
\let\center\flushleft
\let\endcenter\endflushleft
\maketitle
\endgroup

% ######################################################################################################################
%         Supplement Text
% ######################################################################################################################

% \section*{Overview}
% \label{supp:sec:SoftwareComparison}

\vspace*{-2.5em}

This document describes our assessment of pool-sequencing-specific equations for population genetic measures of diversity (such as $\theta_\pi$, $\theta_\text{Watterson}$, Tajima's D), and differentiation (such as \fst{}). 
We re-render some approaches originally presented and implemented in \toolname{PoPoolation} \cite{Kofler2011a} and \toolname{PoPoolation2} \cite{Kofler2011b}.
The aim of these equations is to correct for biases of pool sequencing, such as limited sample size (number of individuals pooled, or pool size $n$) and limited read size (number of reads obtained from those individuals, or coverage $C$).

This document is largely based on two sources:
\begin{itemize}
  \item The reverse-engineered code of \toolname{PoPoolation} and \toolname{PoPoolation2}. We want this document to represent the equations that are actually computed when running these programs, as we feel that they need a more thorough assessment than what is available in the current literature.
  \item The PoPoolation equations document \texttt{correction\_equations.pdf} as found in their code repository; we provide a copy at \url{https://github.com/lczech/popoolation/blob/master/files/correction_equations.pdf}. This document derives some of the equations implemented, but also contains some more that might be interesting for a deeper understanding of the topic.
\end{itemize}

Lastly, we here introduce novel estimators for \fst{} for pool sequencing data, that correct for both biases described above.

% {\color{purple}{\textbf{DISCLAIMER: This document is currently a draft based on our best assessment and written to the best of our best knowledge, in order to start discussing these topics with the research community. There might be parts in PoPoolation that we missed or misinterpreted, and there might be oversights and errors in our derivations. Please feel free to reach out if you want to engage in the discussion!}}}

{\color{purple}{\textbf{DISCLAIMER: This is a draft document (last updated 2022-02-02) that describes the equations as implemented in PoPoolation, to the best of our knowledge. We aim to make this public to discuss our new implementation and novel methods improving on PoPoolation with its developers and the research community. Please reach out to us if you have comments or feedback.}}}

% {\color{purple}{\textbf{Moi: I would like to rephrase  "as we understand them to the best of our knowledge", as it suggests we do not understand them}}}

% ======================================================================================================
%          Definitions
% ======================================================================================================

\section{Definitions}
\label{supp:sec:Definitions}

% We here assume basic familiarity with pool sequencing concepts.
% Please refer to \citeay{Kofler2011a} for details on pool sequencing data and its biases.
% \todo{maybe we should cite some more sources here?}

% ------------------------------------------------------------------------------------------------------
%          Pool Sequencing
% ------------------------------------------------------------------------------------------------------

\subsection{Pool Sequencing Data}
\label{supp:sec:Definitions:sub:PoolSequencing}

We first define the input that we assume to be given for all subsequent equations.
In the software implementation of the equations, these are be based on the input data, or set by the user as parameters.

$n$ : 
Pool size, provided by the user. This is the number of individuals that were pooled together for sequencing.

$C$ : 
Observed coverage. This is the number of reads sequenced from the pool that span the given position in the genome.

$b$ : 
Minimum allele count, provided by the user. 
We do not want to consider SNPs with fewer than $b$ alternative reads in the data, as they might be sequencing errors.
Note that we assume $b$ to be a user-provided constant, 
and hence leave it out of (most) function arguments for simplicity.

% ------------------------------------------------------------------------------------------------------
%          Notation
% ------------------------------------------------------------------------------------------------------

\subsection{Notation}
\label{supp:sec:Definitions:sub:Notation}

$\tau$ : 
Nucleotides, with $\tau \in \left\{ \text{A}, \text{C}, \text{G}, \text{T} \right\}$.

$c_\tau$ : 
Nucleotide counts, \ie, how many reads have a certain nucleotide $\tau$ at a given genomic position. 
Hence, $C = \sum_\tau c_\tau$.

$\bm{c}$ : 
Vector of nucleotide counts (for convenience), 
\ie, $\bm{c} = (~ c_\text{A}, c_\text{C}, c_\text{G}, c_\text{T} ~)$.

% $C$ : total coverage, \ie, the sum of all nucleotide counts, with $C = \sum_\tau c_\tau$

$f_\tau$ : 
Nucleotide frequencies, \ie, $f_\tau = c_\tau / C$. %\sfrac{c_\tau}{C}$ 

$\bm{f}$ : 
Vector of nucleotide frequencies (for convenience), 
\ie, $\bm{f} = (~ f_\text{A}, f_\text{C}, f_\text{G}, f_\text{T} ~)$.

$u$, $v$ : 
For biallelic SNP positions, we simplify, and instead of the four $c_\tau$ values, just use $u$ for the count of the reference (major, or ``first'') allele, and $v$ for the count of the alternative (minor, or ``second'') allele.
We here leave it open whether the reference allele is defined by some reference genome, or simply the major (highest count) allele. \todo{does this make a difference? do the results differ?}
In the PoPoolation notation, this means that $u \equalhat m$, or sometimes $u \equalhat i$; PoPoolation uses both, depending on context, and $v \equalhat C-m$, or $v \equalhat C-i$.

% $p$, $q$ : 
% For biallelic SNP positions, we simplify and instead of the four $f_\tau$ values,
% just use $p$ for the frequency of the reference allele, and $q$ for the frequency of the alternative allele.

$m$ :
Index of summation over potential levels of coverage $C$.

$k$ :
Index of summation over potential pool sizes $n$.

% $a$: (generalized) harmonic numbers

% $h$ : heterozygosity

% ------------------------------------------------------------------------------------------------------
%          Harmonic Numbers
% ------------------------------------------------------------------------------------------------------

\subsection{Harmonic Numbers}
\label{supp:sec:Definitions:sub:HarmonicNumbers}

We define $a_1$ and $a_2$ based on (generalized) harmonic numbers,
as the sum of (squared) reciprocals of the first $n-1$ positive integers:
%
\begin{align}
    \label{eq:an}
    a_1(n) &= \sum_{k=1}^{n-1} \frac{1}{k}
    \\
    \label{eq:bn}
    a_2(n) &= \sum_{k=1}^{n-1} \frac{1}{k^2}
\end{align}
%
These will be needed in several of the below equations.
We use this notation as a compromise between Equation (3.6) of \citeay{Hahn2018} and the notation of $a_n$ and $b_n$ used in \citeay{Achaz2008} for these quantities.
Note that the standard definition of harmonic numbers $H(n)$ includes the $n$-th element, which the above definition does not.

% ======================================================================================================
%          Theta Pi
% ======================================================================================================

\section{Theta Pi}
\label{supp:sec:ThetaPi}

First, we derive equations for $\theta_\pi$, also called Tajima's $\pi$, based on its original (classic) definition, but correcting for biases introduced by pool sequencing, following PoPoolation.

We have two aims. First, to produce an unbiased estimator of $\theta_{\pi}$ based on pool size and coverage-induced noise, and second, to derive an expectation of the ``population mutation rate'' $\theta$ from the former estimator.

% ------------------------------------------------------------------------------------------------------
%          Unbiased Pool-seq estimator
% ------------------------------------------------------------------------------------------------------

% \subsection{Unbiased Pool-seq estimator of \texorpdfstring{$\theta_{\pi}$}{Theta Pi} for biallelic SNPs}
% \label{supp:sec:ThetaPi:sub:PoolSequencing}

\subsubsection*{Unbiased Pool-seq estimator of \texorpdfstring{$\theta_{\pi}$}{Theta Pi} for biallelic SNPs}
\label{supp:sec:ThetaPi:sub:PoolSequencing}

% % ------------------------------------------------------------------------------------------------------
% %     Definition for biallelic SNPs
% % ------------------------------------------------------------------------------------------------------
%
% \subsubsection*{Definition for biallelic SNPs}
% \label{supp:sec:ThetaPi:sub:PoolSequencing:sub:BasicDefinition}

We first define $\theta_\pi$, as usual, as the heterozygosity based on the allele frequencies at a given locus:
%
\begin{align}
    \label{eq:ThetaPiFreqF}
    \theta_\pi(\bm{f}) &= \frac{n}{n-1} \left(1 - \sum_\tau f_\tau^2 \right)
\end{align}
%
using the possible nucleotide frequencies $\bm{f} = (~ f_\text{A}, f_\text{C}, f_\text{G}, f_\text{T} ~)$, with $\tau \in \left\{ \text{A}, \text{C}, \text{G}, \text{T} \right\}$, with Bessel's correction for the number of sequences $n$ (which is the equivalent of individual sequencing for our pool size $n$).
See for example Equation (3.1) of \citeay{Hahn2018} for the original definition for individuals.

In the pool-sequencing case, however, we have a coverage of $C$ reads at a given position in the genome.
We hence can use nucleotide counts $c_\tau$ instead, that is, $f_\tau = c_\tau / C$.
We furthermore use Bessel's correction $\frac{C}{C-1}$ based on the read coverage to obtain an unbiased estimate, and reformulate the above as: 
%
\begin{align}
    \label{eq:ThetaPiFreq2}
    \theta_\pi(\bm{c}, C) &= \frac{C}{C-1} \left(1 - \sum_\tau \frac{c_\tau^2}{C^2} \right) 
\end{align}
%
At this point, the PoPoolation equations document begins to simplify the above equation, and then breaks it down for biallelic SNPs. However, their (and our) implementation differ from this, and use the above equation that can work with any (not just biallelic) SNPs. We hence do not introduce these simplifications here.
\todo{I thought a bit more about this. the heterozygosity is indeed computed from all four bases. BUT: the below equations are still assuming biallelic SNPs! is that a problem?! it's solvable by just using biallelic snps as in the computation -- but what would it mean theoretically to use this correction for non-biallelic snps?}
Note however that the overall computation is still only conducted on biallelic sites, as the correction term introduced below assumes this.

\todo{(Moi) I think it is fine that below it assumes biallelic SNPs. It is a theoretical derivation of 4Neu, one of which assumptions is the infinitesimal model (a mutation is unlikely to happen two times in the same place, so multiallelic loci are assumed impossible)}

\popoolissue{We think that it is makes sense to compute only on biallelic SNPs, as this is a theoretical derivation of $4N_e\mu$, which assumes the infinitesimal model (which does not allow for multi-allelic loci). Still, we briefly wanted to check in with you here. Our guess is that the computation in code is just done for all four nucleotides to keep the code simple -- but the function only gets called for biallelic sites, so it behaves as if it was only computed for biallelic SNPs. Is that right?}

% We can then begin simplifying this equation for easier computation:

% \begin{align}
%     \nonumber
%     \theta_\pi(\bm{c}, C) &= \frac{C}{C-1} \left(1 - \sum_\tau \frac{c_\tau^2}{C^2} \right) 
%     \\
%     \nonumber
%     &= \frac{C}{C-1} - \sum_\tau \frac{c_\tau^2}{C(C-1)} 
%     \\
%     \nonumber
%     &= \frac{C^2}{C(C-1)} - \sum_\tau \frac{c_\tau^2}{C(C-1)} 
%     \\
%     \intertext{
%         Now, we use $C = \sum_\tau c_\tau$ to extend the numerators:
%     }
%     \nonumber
%     &= \frac{C^2 - C}{C(C-1)} - \sum_\tau \frac{c_\tau^2 - c_\tau}{C(C-1)} 
%     \\
%     \nonumber
%     &= \frac{C(C-1)}{C(C-1)} - \sum_\tau \frac{c_\tau(c_\tau - 1)}{C(C-1)} 
%     \\
%     \label{eq:ThetaPiFreq3}
%     &= 1 - \sum_\tau \frac{c_\tau(c_\tau - 1)}{C(C-1)}
% \end{align}

% For a simple biallelic SNP, we only have two counts with $C = u+v$ instead of four $c_\tau$ counts.
% Substituting this in \eqnref{eq:ThetaPiFreq2}, we get:

% \begin{align}
%     \nonumber
%     \theta_\pi(u,v,C) &= \frac{C}{C-1} \left(1 - \frac{u^2}{C^2} - \frac{v^2}{C^2} \right) 
%     \\
%     \nonumber
%     &= \frac{C^2}{C(C-1)} \left(1 - \frac{u^2}{C^2} - \frac{v^2}{C^2} \right) 
%     \\
%     \nonumber
%     &= \frac{C^2}{C(C-1)} - \frac{u^2}{C(C-1)} - \frac{v^2}{C(C-1)} 
%     \\
%     \intertext{
%         This contains redundant information; let's further simplify using $v = C-u$:
%     }
%     \nonumber
%     \theta_\pi(u,C) &= \frac{C^2}{C(C-1)} - \frac{u^2}{C(C-1)} - \frac{(C-u)^2}{C(C-1)}
%     \\
%     \nonumber
%     &= \frac{C^2 - u^2 - (C-u)^2}{C(C-1)}
%     \\
%     \nonumber
%     &= \frac{(C+u)(C-u) - (C-u)^2}{C(C-1)}
%     \\
%     \nonumber
%     &= \frac{\left[(C+u) - (C-u)\right](C-u)}{C(C-1)}
%     \\
%     \label{eq:ThetaPiSimple}
%     &= \frac{2u(C-u)}{C(C-1)}
% \end{align}

% This is the final equation for $\theta_\pi$ estimator for a biallelic SNP that we will use for pool sequencing, 
% with a coverage of $C$, composed of two SNP counts with $u$ as the first and $v$ as the second allele count, and $u+v=C$.

%If desired, this final estimator equation can be corrected for small pools of $n$ individual:
%
%\begin{align}
%    \theta_\pi(u,v,C,n) &= \frac{n}{n-1}  %\frac{2u(C-u)}{C(C-1)}
%\end{align}
%
%\todo{the above is new. Moi, did you add this? should we implement this?}

% ------------------------------------------------------------------------------------------------------
%    Expected value of population mutation rate $\theta$ from nucleotide diversity
% ------------------------------------------------------------------------------------------------------

\subsubsection*{Expected value of population mutation rate $\theta$ from nucleotide diversity}
\label{supp:sec:ThetaPi:sub:PoolSequencing:sub:ExpectedValue}

\todo{we probably need to explain better what we mean by expected  $\theta_\pi$. Jeff, would it be more appropriate to have something like $\theta_{\theta_\pi}$ to be clear? or $\hat{\theta_\pi}$ ? }

We now use the expectation of the $\theta_\pi$ estimator to infer the population mutation rate $\hat{\theta}$, accounting for noise generated by the total coverage $C$ and pools of $n$ individuals. We assume a biallelic site, and use expectations from a neutral site frequency spectrum under constant population size:
%
\begin{align}
\label{eq:ExpectationThetaPi}
\mathbb{E}(\theta_\pi|C,n) &= P(\mbox{SNP} | n) \cdot \sum_{m=b}^{C-b} \theta_\pi(m,C) \cdot P(m|C,n)
% \mathbb{E}(\theta_\pi|C) &= P(\mbox{SNP} | n) \cdot \sum_{m=b}^{C-b} \frac{2m (C-m)}{C(C-1)} \cdot P(m|C,n)
% \\
% &= P(\mbox{SNP} | n) \cdot \frac{2}{C(C-1)} \cdot \sum_{m=b}^{C-b} m(C-m) \cdot P(m|C,n)
\end{align}
%
In words, the expected value is computed by summing all possible SNP counts (that exceed the minimum count $b$) 
that can occur in a pool with coverage $C$ 
(using the first/major allele count $m$ here, with second/minor allele count $C-m$ implicit),
weighted by the probability to have each of those counts, %in the first place, 
and scaled by the probability to have a SNP in the first place. 

Here, we are using the minimum allele count $b$ that we want to consider (as provided by the user),
meaning that we only consider SNPs that have at least $b$ reads for either the first or second allele. %$u$ or $v$. 
As we are only using the first allele count $m$ in the equation above, 
and do not know which of the two counts is the larger one, 
we ``sandwhich'' our potential values for the coverage between $b$ and $C-b$.

The two probabilities used above are computed as follows.

$P(\mbox{SNP} | n)$ is the probability of observing a SNP in a pool of $n$ individuals:
%
\begin{align}
\label{eq:PSNP}
P(\mbox{SNP} | n) &= \theta \sum_{k=1}^{n-1} \frac{1}{k} = \theta  a_1(n)
\end{align}
%
Assuming that all variation is neutral, and that the population is of constant size and in mutation-drift equilibrium, 
by definition, $\theta = \mathbb{E}( S / a_1(n) )$ with $S$ segregating sites.
Then, the $a_1$ terms cancel out, meaning that \eqnref{eq:PSNP} yields the proportion of variable sites.

$P(m|C, n)$ is the probability of observing $m$ as first allele count in a SNP with $C$ reads from a pool of dimension $n$:
%
\begin{align}
    \label{eq:Pmcn}
    P(m|C,n) 
    %&= \sum_{k=1}^{n-1} P(m|C,n,k) \frac{1/k}{\sum_{j=1}^{n-1} 1/j} \\
    &= \frac{1}{a_1(n)}  \sum_{k=1}^{n-1} \frac{1}{k}  P(m|C,n,k)
\end{align}
%
$P(m|C, n, k)$ is the probability of having a first allele count of $m$ observed in $C$ reads that were taken
from a pool of $n$ individuals with first allele count of $k$.
That is, $m$ is the allele count in the reads, and $k$ is the allele count in the pool:
%
\begin{align}
    \label{eq:Pmcnk}
    P(m|C,n,k) &= {C \choose m} \left(\frac{k}{n}\right)^m \left(\frac{n-k}{n}\right)^{C-m}
\end{align}
%
In words, $P(m|C, n, k)$ follows a binomial distribution, with $m$ successes out of $C$ trials 
with a success probability of $k/n$ for each trial.
That is, we compute how likely it is to observe $m$ as the first/major allele count in $C$ reads, 
given the frequency $k/n$ of the major allele in the pool.
Again, the count of the second/minor allele is implicitly used here as $C-m$.

Starting from \eqnref{eq:ExpectationThetaPi}, we can now put this together:
%
\begin{align}
\nonumber
\mathbb{E}(\theta_\pi|C,n) &= P(\mbox{SNP} | n) \cdot \sum_{m=b}^{C-b} \theta_\pi(m,C) \cdot P(m|C,n)
\\
% \nonumber
\label{eq:ExpThetaPiLong}
&= \theta  a_1(n) \cdot \sum_{m=b}^{C-b} \frac{2u(C-m)}{C(C-1)} \cdot \frac{1}{a_1(n)}  \sum_{k=1}^{n-1} \frac{1}{k}  {C \choose m} \left(\frac{k}{n}\right)^m \left(\frac{n-k}{n}\right)^{C-m}
\end{align}

%@\todo{Jeff, Lucas I think there is still something weird in the substitution of the P(SNP given n). As it is written using Theta= S/an, the substitution is not a probability but the number of segregating sites}


% ------------------------------------------------------------------------------------------------------
%     Final approximation for population mutation rate Theta
% ------------------------------------------------------------------------------------------------------

\subsubsection*{Final approximation for population mutation rate Theta}
\label{supp:sec:ThetaPi:sub:PoolSequencing:sub:FinalApprox}

We can now solve this for $\theta$ to define our final corrected estimate $\theta_{\pi,\text{pool}}$. 
%
\begin{align}
    \label{eq:ThetaPiPoolEstimate}
    \theta &\approx
    \frac{ 
        \mathbb{E}(\theta_\pi|C,n)
    }{ 
        a_1(n) \cdot \sum_{m=b}^{C-b} \theta_\pi(m,C) \cdot P(m|C,n)        
%         \sum_{k=1}^{n-1} \frac{1}{k}  {C \choose m} \left(\frac{k}{n}\right)^m \left(\frac{n-k}{n}\right)^{C-m} 
    }
\end{align}
%
This only leaves the $\mathbb{E}(\theta_\pi|C,n)$ term unresolved, % (the left hand side of \eqnref{eq:ExpThetaPiLong}),
which we however can estimate from our data using the classic estimator as shown in 
\eqnref{eq:ThetaPiFreq2};
% \eqnref{eq:ThetaPiSimple};
note however that this is only evaluated on biallelic SNPs that have at least a count of $b$.
In total, this yields:
%
\begin{align}
    \label{eq:ThetaPiPool}
    \theta_{\pi,\text{pool}}(\bm{c},C,n) &:=  
    \frac{ 
        \frac{C}{C-1} \left( 1 - \sum_\tau \frac{c_\tau^2}{C^2} \right)
    }{ 
        \sum_{m=b}^{C-b} \frac{2m(C-m)}{C(C-1)} \cdot 
        \sum_{k=1}^{n-1} \frac{1}{k}  {C \choose m} \left(\frac{k}{n}\right)^m \left(\frac{n-k}{n}\right)^{C-m} 
    }
\end{align}
%
Note that the $a_1$ terms cancel out, and 
that the denominator only depends on the total coverage $C$ and the pool size $n$ (and not on any allele counts $c_\tau$),
and hence only needs to be computed once per coverage level, yielding a significant computational speedup.

The above computation is for a single variant site.
For a whole region or window, the values of $\theta_{\pi,\text{pool}}(\bm{c},C,n)$ are simply summed up.
% That is, we are using a sum of ratios here.
This is the equation as implemented in \toolname{PoPoolation} as the measure called \texttt{pi},
and implemented in our \toolname{grenedalf} as well.

\todo{Jeff, is this all sound now? Is this the correct estimator for what we want it to do?}
\todo{(Moi) I think so, this was already checked)}


% \begin{align}
%  \mathbb{E}(\theta_\pi,c) = \theta a_1(n) \frac{C}{C-1}\sum_{m=b}^{C-b}\left( 1 - 
% \left(\frac{m}{C}\right)^2 \right) \cdot \sum_{k=1}^{n-1} {C \choose m} \left(\frac{k}{n}\right)^m \left(\frac{n-k}{n}\right)^{C-m}  \frac{1}{k} \frac{1}{a_1(n)}
% \end{align}
% 
% \begin{align}
% \mathbb{E}(\theta_\pi|C) &= P(\mbox{SNP} | n) \cdot \sum_{m=b}^{C-b} \frac{2m (C-m)}{C(C-1)} \cdot P(m|C,n)
% \\
% &= \theta \cdot a_1(n) \cdot \sum_{m=b}^{C-b} \frac{2m (C-m)}{C(C-1)} \cdot \frac{1}{a_1(n)} \sum_{k=1}^{n-1} \frac{1}{k} P(m|C,n,k) \\
% &= \theta \cdot a_1(n) \cdot \sum_{m=b}^{C-b} \frac{2m (C-m)}{C(C-1)} \cdot \frac{1}{a_1(n)} \sum_{k=1}^{n-1} \frac{1}{k} {C \choose m} \left(\frac{k}{n}\right)^m \left(\frac{n-k}{n}\right)^{C-m}
% \end{align}

% \clearpage

% ------------------------------------------------------------------------------------------------------
%          Computation in Windows
% ------------------------------------------------------------------------------------------------------

% \subsection{Computation in Windows}
% \label{supp:sec:ThetaPi:sub:Windows}

% not much to say here, so moved to above

% % ------------------------------------------------------------------------------------------------------
% %          Simplified Theta Pi
% % ------------------------------------------------------------------------------------------------------

% \subsection{Simplified Theta Pi}
% \label{supp:sec:ThetaPi:sub:Simplified}

% Not sure if needed.

% PoPoolation notes: also good for individual sequencing

% ======================================================================================================
%          Theta Watterson
% ======================================================================================================

\section{Theta Watterson}
\label{supp:sec:ThetaWatterson}

% ------------------------------------------------------------------------------------------------------
%          Theta Watterson
% ------------------------------------------------------------------------------------------------------

% \subsection{Pool-Sequencing Correction}
% \label{supp:sec:ThetaWatterson:sub:PoolSequencing}

For Watteron's estimator $\theta_w$, we follow the same approach as above.
In order to derive the pool-sequencing corrected equations, we first define $\theta_w$ as usual:
% 
\begin{align}
    \label{eq:ThetaWClassic}
    \theta_w(u, C) &= \frac{S(u)}{ \sum_{k=1}^{C-1} 1/k }
\end{align}
% 
where classically, $S$ is the number of segregating sites, see for example Equation (3.5) of \citeay{Hahn2018}.
We are here working with a biallelic SNP at a single site, 
which as before we only want to consider if its count is within the limits of the minimum allele count $b$, 
% that is, within $b$ and $C-b$, 
and so we define:
% 
\begin{align}
%     \nonumber
    S(u) = 
    \begin{cases}
        1 & \text{if } b \le u \le C-b
        \\
        0 & \text{otherwise}
    \end{cases}
\end{align}
% 
Reasoning the same as above, we get the expected value of $\theta_w$ as:
% 
\begin{align}
    \nonumber
    \mathbb{E}(\theta_w|C,n) 
    &= P(\mbox{SNP} | n) \cdot \frac{
        \sum_{m=b}^{C-b} P(m|C,n)
    }{
        \sum_{k=1}^{C-1} 1/k
    }
    \\
    \intertext{
        with the two probability terms again as in \eqnref{eq:PSNP} and \eqnref{eq:Pmcn}.
        For conciseness, we here only resolve $P(\mbox{SNP} | n)$:
    }
    \label{eq:ExpecationThetaW}
    &= \theta  a_1(n) \cdot \frac{
        \sum_{m=b}^{C-b} P(m|C,n)
    }{
        \sum_{k=1}^{C-1} 1/k
    }
\end{align}
% 
We can again solve this for $\theta$, to get our corrected estimate:
%
\begin{align}
    \label{eq:CorrectedThetaEstimate}
    \theta &\approx 
    \mathbb{E}(\theta_w|C,n) \cdot 
    \frac{
        \sum_{k=1}^{C-1} 1/k
    }{
        a_1(n) \cdot \sum_{m=b}^{C-b} P(m|C,n)
    }
\end{align}
%
Again we can use the classic value $\theta_w(u, C)$ of \eqnref{eq:ThetaWClassic} for the expected value $\mathbb{E}(\theta_w|C,n)$, 
so that the summation over $1/k$ in the numerator and in $\theta_w(u, C)$ cancel out here.
We can now define our estimate:
%
\begin{align}
    % \label{eq:ThetaWPoolEst}
    \nonumber
    \theta_{w,\text{pool}}(u,C,n) 
    &:= \frac{
        S(u)
    }{
        a_1(n) \cdot \sum_{m=b}^{C-b} P(m|C,n)
    } \\
    \intertext{
        We can now replace $P(m|C,n)$ according to \eqnref{eq:Pmcn} and \eqnref{eq:Pmcnk}. The $a_1(n)$ terms in the denominator and in $P(m|C,n)$ cancel out, leading to the final equation:
    }
    \nonumber
    &= \frac{S(u)}{ \sum_{m=b}^{C-b} \sum_{k=1}^{n-1} \frac{1}{k} P(m|C,n,k) } \\
    \label{eq:ThetaWPool}
    &= \frac{S(u)}{ \sum_{m=b}^{C-b} \sum_{k=1}^{n-1} \frac{1}{k} {C \choose m} \left(\frac{k}{n}\right)^m \left(\frac{n-k}{n}\right)^{C-m} }
\end{align}
%
As before, the denominator only depends on the coverage $C$,
and hence only needs to be computed once per coverage level that is present in the data.

Again, the approach to compute this for a window is to sum up all values across the SNPs in the window.
This is the equation as implemented in \toolname{PoPoolation} as the measure called \texttt{theta},
and implemented in our \toolname{grenedalf} as well.

\todo{Jeff, is this all sound now? Is this the correct estimator for what we want it to do?}
\todo{(Moi) I think Jeff wenth through this already}

% ------------------------------------------------------------------------------------------------------
%          Simplified Theta Watterson
% ------------------------------------------------------------------------------------------------------

% \subsection{Simplified Theta Watteron}
% \label{supp:sec:ThetaWatterson:sub:Simplified}

% Not sure if needed.

% PoPoolation notes: also good for individual sequencing

% ======================================================================================================
%          Tajima's D
% ======================================================================================================

\section{Tajima's D}
\label{supp:sec:TajimaD}

Above, we have defined pool-sequencing corrected estimators $\theta_\pi$ and $\theta_w$.
Now, we want to use them to define a test akin to Tajima's D for pool sequencing.
We are here again following the PoPoolation approach, and re-derive their equations.

% ------------------------------------------------------------------------------------------------------
%          Pool-Sequencing Correction
% ------------------------------------------------------------------------------------------------------

\subsection{Pool-Sequencing Correction}
\label{supp:sec:TajimaD:sub:PoolSequencingCorrection}

The PoPoolation equations document derives the following estimator.
To the best of our knowledge, this is however not implemented in \toolname{PoPoolation};
instead, they compute Tajima's D as presented in the following \secref{supp:sec:TajimaD:sub:Classic}.
We still introduce the approach here, for reference, and in the hope that it might be helpful.

First, we define:
% 
\begin{align}
    d_\text{pool}(u,C) ~&:=~ \theta_{\pi,\text{pool}}(u,C) ~-~ \theta_{w,\text{pool}}(u,C)
\end{align}
% 
using the major allele count $u$ at a site with coverage $C$, and use this to define our statistic:
% 
\begin{align}
    D_\text{pool}(u,C) ~&:=~ \frac{ d_\text{pool}(u,C) }{ \sqrt{ \text{Var}( d_\text{pool}(u,C) ) }}
\end{align}
% 
In order to compute the variance of $d_\text{pool}$ (leaving out function arguments for simplicty), 
we start with the standard expansion of the variance:
% 
\begin{align}
    \nonumber
    \text{Var}( d_\text{pool} ) &= \mathbb{E}( d_\text{pool}^2 ) - \mathbb{E}(d_\text{pool})^2
\end{align}
% 
At this point, we use that $d_\text{pool}$ is unbiased (for populations at equilibrium) \todo{Jeff, is it really?!  Jeff says: Should be for equilibrium popualtions, but not for non-equilibrium populations. Lucas answers: Thanks, I added that now!}, and hence has an expected value of $0$, that is, $\mathbb{E}(d_\text{pool})^2 = 0$.
The PoPoolation equations document notes that this is only true if they did their previous calculations correctly, but we trust they did.

Then, we can compute the variance as:
% 
\begin{align}
    \nonumber
    \text{Var}( d_\text{pool} ) &= \mathbb{E}( d_\text{pool}^2 )
    \\
    \nonumber
    &= P(\mbox{SNP} | n) \cdot \sum_{m=b}^{C-b} d_{\text{pool}}^2(m, C) \cdot P(m|C,n)
    \intertext{which can be resolved using equations \eqnref{eq:PSNP} and \eqnref{eq:Pmcn} from previous sections:}
    \label{eq:VarDPool}
    &= \theta  \cdot \sum_{m=b}^{C-b} \left( \theta_{\pi,\text{pool}}(m,C) - \theta_{w,\text{pool}}(m,C) \right)^2 
    \cdot \sum_{k=1}^{n-1} \frac{1}{k}  {C \choose m} \left(\frac{k}{n}\right)^m \left(\frac{n-k}{n}\right)^{C-m}
\end{align}
% 
This leaves $\theta$ to be estimated.
PoPoolation suggests to estimate it as $\theta_{\pi,\text{pool}}$ 
on the same window on which we are computing $D_\text{pool}$.
This assumes that all individuals contribute the same number of reads to the pool.

The first summation in \eqnref{eq:VarDPool} involves computing $\theta_{\pi,\text{pool}}$ and 
$\theta_{w,\text{pool}}$ repeatedly $C-2b$ many times, 
with each of these computations involving to compute their respective denominators,
as shown in \eqnref{eq:ThetaPiPool} and \eqnref{eq:ThetaWPool}.
However, as $C$ remains constant throughout this computation, these denominators (the correction terms)
are identical, so that we only need to compute them once, to gain a $\approx C$-fold speedup.

At this point, the PoPoolation equation document also introduces an approach to compute Tajima's D based on the above in windows.
We here skip this part for brevity.

% \begin{align}
% D_{\text{pool}} &= \frac{ \theta_{\pi,\text{pool}} - \theta_{w,\text{pool}}
% }{ \sqrt{\text{Var}( \theta_{\pi,\text{pool}} - \theta_{w,\text{pool}} ) }} 
% \\
% \text{Var}( \theta_{\pi,\text{pool}} - \theta_{w,\text{pool}} ) &= 
% \\\\
% d_{\text{pool}} &= \theta_{\pi,\text{pool}} - \theta_{w,\text{pool}}
% \\
% D_{\text{pool}} &= \frac{ d_{\text{pool}}}{ \sqrt{\text{Var}( d_{\text{pool}} ) }}
% \\
% \text{Var}( d_{\text{pool}} ) &= \mathbb{E}(d_{\text{pool}}^2) - \mathbb{E}(d_{\text{pool}}) ^2
% \end{align}

% ------------------------------------------------------------------------------------------------------
%          Computation in Windows
% ------------------------------------------------------------------------------------------------------

% \subsection{Computation in Windows}
% \label{supp:sec:TajimaD:sub:Windows}

% \todo{not sure if needed}

% \todo{@Lucas. If implemented, please add}

% ------------------------------------------------------------------------------------------------------
%          Integration with Classic D
% ------------------------------------------------------------------------------------------------------

\subsection{Integration with Classic Tajima's D}
\label{supp:sec:TajimaD:sub:Classic}

On large windows, the classic Tajima's D is not a measure of significance (in number of standard deviations away from the null hypothesis), but instead is a measure of the magnitude of the divergence from neutrality.
This is because all loci are considered completely linked, even if they are not in reality.

However, the above pool-sequencing Tajima's D instead consideres all loci as completely unlinked, \todo{Moi, Jeff, is that the case? (Moi answer) The above method is unlinked, I believe, because the difference is computed per SNP. The classic} and thus represents the number of standard deviations away from neutrality.
Therefore, it gives a different numerical result that has a much higher absolute value compared to classic Tajima's D.

Now, we want to obtain a correction term for the pool-sequence Tajima's D to obtain values that are comparable to classic Tajima's D in non-small windows, that is, we want a measure of the magnitude of the divergence from neutrality.
We again follow the PoPoolation approach, and here derive the equations that are actually implemented.

% ------------------------------------------------------------------------------------------------------
%      Approach by Achaz
% ------------------------------------------------------------------------------------------------------

\subsubsection*{Approach by Achaz}

To this end, PoPoolation2 uses a modified version of the $Y^*$ test of \citeay{Achaz2008},
which was originally developed as a test for neutrality despite the presence of sequencing errors.
This test only works when excluding singletons, that is, we set $b:=2$ for this part.

Following PoPoolation and \citeay{Achaz2008}, we first define:
% 
\begin{align}
f^*(n) &= \frac{n-3}{a_1(n) \cdot (n-1)-n}
% \\
% \nonumber
\intertext{which is then used to define:}
% \\
\alpha^*(n) &= f^{*2} \cdot\left( a_1(n) - \frac{n}{n-1} \right) + f^* \cdot\left( a_1(n) \cdot \frac{4(n+1)}{(n-1)^2} - 2 \cdot\frac{n+3}{n-1} \right) - a_1(n) \cdot\frac{8(n+1)}{n(n-1)^2} + \frac{n^2+n+60}{3n(n-1)}
% \\
% \nonumber
\intertext{and:}
% \\
\nonumber
\beta^*(n) &= f^{*2} \cdot\left( a_2(n) - \frac{2n-1}{(n-1)^2} \right) + f^* \cdot\left( a_2(n) \cdot \frac{8}{n-1} - a_1(n) \cdot\frac{4}{n(n-1)} - \frac{n^3 + 12n^2 -35n +18}{n(n-1)^2} \right) \\
\label{eq:betastar}
&- a_2(n) \cdot\frac{16}{n(n-1)} + a_1(n) \cdot\frac{8}{n^2(n-1)} + \frac{2(n^4 + 110n^2 -255n + 126)}{9n^2(n-1)^2}
\end{align}
% 
Note that these equations were originally developed for data from individuals, 
and hence here, $n$ denotes the number of individuals \emph{as if} we were doing individual sequencing.

NB: The PoPoolation document recommends to counter-check the correctness of their equation 
with the original of \citeay{Achaz2008}. 
In fact, PoPoolation introduced a slight mistake in the last term of $\beta^*$, 
which we have fixed here. Above is the (hopefully) correct one, following \citeay{Achaz2008}.
Note that the mistake only concerns the PoPoolation equations document, but not their implementation.

\popoolissue{The above is indeed not a big one, but we thought it's good to mention it.}

\popoolissue{A more serious issue occurred in the computation of alpha*. The code actually computes this as beta*, and never calls the actual alpha* function, see \href{https://github.com/lczech/popoolation/blob/092e7a6f7ee4910c1bec4377e0adccc353175bc8/Modules/VarMath.pm\#L104}{here}. This is a bug that Robert and I have discussed before, and I think it is fixed now.}

\popoolissue{Another small issue: The computation of alpha* requires ``effective coverage'' to be larger than 1, see \href{https://github.com/lczech/popoolation/blob/092e7a6f7ee4910c1bec4377e0adccc353175bc8/Modules/VarMath.pm\#L226}{here}, but tests this against $n$, which is the number of individuals. Are we missing something here?}

\popoolissue{Furthermore, we tested the computation of alpha*, and it also gives NaN values for input 2 and 3, so the test should in fact be $>= 4$ instead.}

% ------------------------------------------------------------------------------------------------------
%     The number of individuals sequenced
% ------------------------------------------------------------------------------------------------------

\subsubsection*{The number of individuals sequenced}

The only unresolved parameter is $n$, which corresponds to the number of individuals sequenced -- 
if we were to do individual sequencing. 
In our case of pool sequencing, according to PoPoolation, 
we can reasonably substitute this with the expected number of distinct individuals sequenced.

To this end, we use the coverage $C$, as well as the pool size $n$, which we here use as our substitute for the number of individuals sequenced.
% \todo{I have no idea what this distinction between $n$ and $n$ is about, and the PoPoolation document does not explain this at all... It might be that $n$ is meant to be the $n$ (pool size) from previous sections, but $n$ here is used as the parameter for $f^*$ instead?!}
Then, we define $\tilde{n}$ (called $\tilde{n}_{base}$ in the PoPoolation equations document) as the expected number of individuals from our pool that have been sequenced:
% 
\begin{align}
    \label{eq:IndivSeqBruteForce}
    \tilde{n}
%     &= \sum_{k=1}^{\text{max}(C, n)} k \cdot \frac{{n \choose k} \sum_{j=1}^{k} (-1)^{k-j} {k \choose j} j^C }{n^C}
    &= \sum_{k=1}^{T} \sum_{j=1}^{k} (-1)^{k-j} \cdot k {n \choose k}  {k \choose j} \left(\frac{j}{n}\right)^C
\end{align}
% 
where $T=\text{max}(C, n)$; 
\todo{max according to the PoPoolation equations document, but min according to their code... yet another bug?! it's even named min the code, so it seems they knew what they were doing there... utterly confusing}
if $n$ is much larger than $C$, we can assume $\tilde{n} \approx C$. 
\todo{The following sentence comes from their equations document, but this is not what they implemented. see also my comment below -- this might be yet another bug}
Our substitute $\tilde{n}$ is then obtained by averaging $\tilde{n}$ over the window $W$.

\popoolissue{We think that there is either an error here, or a bug in the code. The code in PoPoolation actually uses the minimum of coverage and pool size in the above, instead of the maximum as stated above, see \href{https://github.com/lczech/popoolation/blob/092e7a6f7ee4910c1bec4377e0adccc353175bc8/Modules/VarMath.pm\#L156}{here}. We are not sure which one is correct though, or what the implications of this are.}

\popoolissue{Furthermore, the original equations document states that ``$\tilde{n}$ is obtained averaging $\tilde{n}$ over the window W'', which we also stated above. However, looking at the code \href{https://github.com/lczech/popoolation/blob/092e7a6f7ee4910c1bec4377e0adccc353175bc8/Modules/VarMath.pm\#L127}{here}, there is no averaging over the window, and instead it's just computed once. The function for averaging exists in the same PoPoolation function as well, see \href{https://github.com/lczech/popoolation/blob/092e7a6f7ee4910c1bec4377e0adccc353175bc8/Modules/VarMath.pm\#L117-L125}{here}, but is commented out, and also contains a bug, as \texttt{\$cov} is not used. We are unsure how much averaging over the window vs just one value actually changes the results, and hence wanted to bring it up here.} 

Computing the expected number of distinct individuals sequenced corresponds to the following statistical question: 
Given a set of integers $A = \{1, \ldots, n\}$ (corresponding to individuals), 
pick a set $B$ of $C$ elements from set $A$ with replacement (corresponding to reads);
what is the expected number of distinct values (individuals) that have been picked in $B$ (that we have reads from)?

PoPoolation computes this value by brute force using \eqnref{eq:IndivSeqBruteForce}, 
that is, by trying all possible ways to pick numbers from the set.
However, there exists a closed form solution to this question, which yields massive speedups for larger coverages, which we have implemented.

One way to arrive at the closed form expression is as follows:
% https://math.stackexchange.com/questions/5775/how-many-bins-do-random-numbers-fill
Define an indicator random variable $I_i$ for $1 \leq i \leq n$ as $1$ 
if individual $i$ is present in the set $B$ (that is, if individual $i$ has been sequenced), and as $0$ if not. 
Then, the size of set $B$ is simply $\sum_{i=1}^{n} I_i$. 

The probability that $I_i$ equals $1$ (that is, that individual $i$ has been sequenced) 
for any $i$ is given by:
% 
\begin{align}
    P(I_i = 1) &= 1-\left( \frac{n-1}{n} \right)^C
\end{align}
% 
In words, this is the complement of \emph{not} picking $i$ in all of the $C$ picks from set $A$.

The expected size of the set $B$ can then be computed by linearity of expectation for all $i$,
yielding our closed form expression:
% 
% $n \left[ 1- \left( 1 - \frac{1}{n} \right)^n \right]$.
% Hence, we compute $\tilde{n}$ as:
% 
\begin{align}
    \tilde{n} = n \left( 1 - \left( \frac{n-1}{n} \right)^C \right)
\end{align}
% 
This is equation that we compute in our implementation to arrive at $\tilde{n}$ for a given coverage $C$ and poolsize $n$.

\popoolissue{Here seem to be more issues in the PoPoolation code. Looking again at the line where the call to compute $n_\text{base}$ is located, \href{https://github.com/lczech/popoolation/blob/092e7a6f7ee4910c1bec4377e0adccc353175bc8/Modules/VarMath.pm\#L127}{here}, the function is called with the min coverage, instead of the actual coverage as stated in the equations here. We are not sure what effect this has, but think that this is an issue.}

\popoolissue{Furthermore, that call to \texttt{nbase\_buffer} takes two arguments, \texttt{\$n} and \texttt{\$mincoverage}, but the inner function of \texttt{get\_nbase\_buffer} only does one \texttt{shift}, hence ignoring the second argument. Robert changed that line after our conversation, but I am not sure that everything is fixed yet, see below.}
 
\popoolissue{At this point also, see \href{https://github.com/lczech/popoolation/blob/092e7a6f7ee4910c1bec4377e0adccc353175bc8/Modules/VarMath.pm\#L145}{here}, the variable is again called \texttt{\$cov}, indicating that we want to take the coverage (and not the min coverage). But because of the missing shift, this value is in fact the pool size again, from the call to the inner function mentioned above. Unless we are misunderstanding perl function call rules (which might well be the case), there seems to be something really off here. In summary: The function is computed with coverage=poolsize, but with a detour that first also makes it seem that coverage=min coverage by accident.}

\popoolissue{Lastly, we noted that the computation of $\tilde{n}$ generally results in some non-integer value. However, if we understand the equations for alpha* and beta* correctly, they expect integer values. We did some quick test of this, and it seems that some low non-integer values of pool sizes and coverages can even lead to negative results then. This should not be an issue in practice, but it still seems that something is not quite right here.}

\todo{this computed using the coverage $C$ in the above equations (and in the PoPoolation equations), but seems to be called with $b$ (min coverage) in their code (and in mine). Jeff, is that yet another bug (number three then, on top of the two already mentioned two sections below) that they introduced in their code? it seems weird to compute this based on the min coverage, than the actual coverage at the site}

\todo{this also leads to another confusing statement in the PoPoolation equations: ``And $\tilde{n}$ is obtained averaging $\tilde{n}$ over the window W.'' -- they never seem to compute the average, and instead just compute it once using the min coverage. I think that is bug number four then, or wrong in their equations document. the function for computing the average exists in their code, but is commented out, not used, and even if it were used, it has yet another bug of its own...}

\todo{Jeff. the nbase value yields non integer results (it's an expected number, so it is not necessarily an integer), however the alpha star and beta star seem to expect integers. is that a problem? computationally, that's fine, but i'm wondering what that means in theory for the equations}

% ------------------------------------------------------------------------------------------------------
\subsubsection*{Final estimator for D}
% ------------------------------------------------------------------------------------------------------

Now that we have a way of computing a reasonable value for the number of individuals sequenced,
we can finally define the estimator:
% 
\begin{align}
    \tilde{D}_\text{pool} &= 
    \frac{
        \theta_\pi - \theta_w
    }{
        \sqrt{ {|W|}^{-1} \cdot \alpha^*(\tilde{n}) \cdot \theta ~+~ \beta^*(\tilde{n}) \cdot  \theta^2 }
    }
\end{align}
% 
% \todo{theta b pool W?!}
following PoPoolation and \citeay{Achaz2008}.
This requires $b=2$; furthermore, PoPoolation suggests to use ``not too small'' windows.
We are using the size $|W|$ of the window here, that is, the total length along the genome, which is typically much larger than the number of SNPs in the window \todo{Jeff, that is what i get from their code. is that the correct term to use here? should it be the number of SNPs in the window instead?}.
The $\theta$ used in the denominator is simply $\theta_w$ again.

The above is the estimator as implemented in \toolname{PoPoolation} and in our implementation.

% https://math.stackexchange.com/questions/72223/finding-expected-number-of-distinct-values-selected-from-a-set-of-integers

% \begin{align}
% &\sum_d\sum_k(-1)^{d-k}d\binom{n}{d}\binom{d}{k}\left(\frac{k}{n}\right)^p\\
% &=\sum_d\sum_k(-1)^{d-k}d\binom{n}{k}\binom{n-k}{n-d}\left(\frac{k}{n}\right)^p\\
% &=n-\sum_d\sum_k(-1)^{d-k}(n-d)\binom{n}{k}\binom{n-k}{n-d}\left(\frac{k}{n}\right)^p\\
% &=n-\sum_d\sum_k(-1)^{d-k}(n-k)\binom{n}{k}\binom{n-k-1}{n-d-1}\left(\frac{k}{n}\right)^p\\
% &=n-\sum_k(n-k)\binom{n}{k}\delta(n-k-1)\left(\frac{k}{n}\right)^p\\
% &=n-n\left(\frac{n-1}{n}\right)^p\\
% &=n\left(1-\left(\frac{n-1}{n}\right)^p\right)\tag{2}
% \end{align}

% ------------------------------------------------------------------------------------------------------
%          Assumptions and Biases
% ------------------------------------------------------------------------------------------------------

\subsection{Assumptions and Biases}
\label{supp:sec:TajimaD:sub:AssumptionsBiases}

\todo{Who wrote that part? I (Lucas) cannot remember it... is this all right? Do we keep it? Which document is the ``PoPoolation manual'' mentioned below? I feel like I have not seen this document yet.}
\todo{(Moi) Lucas, I think you wrote this part based on the PoPoolation repository equations PDF}

In the above computation of the correction term for Tajima's D for pool sequencing,
several assumptions are made that lead to the resulting estimator being conservative,
\ie, yielding smaller values that what would be expected from individual sequencing of samples.
Based on the explanation in the PoPoolation manual (most of the text in this section is copied from there), 
we here explore the underlying assumptions and biases.

The locally fluctuating coverage is replaced by the minimum coverage. 
This makes the variance estimator larger, and therefore leads to conservative estimates of Tajima's D.

The random number of different individuals sequenced under a given coverage $C$ 
is replaced by its expected value $\tilde{n}$. 
This assumption should not affect the results much:
If the pool size is large compared to the coverage, sequencing the same individual more than once is uncommon. 

Furthermore the number of different individuals sequenced will have a low variance. 
As we are working with the minimum coverage, $\tilde{n}$ will be biased downwards,
tending to give a conservative estimate of the variance.

At different positions, the subsets from the pool are sequenced might be different. 
Their coalescent histories will be correlated but not identical. 
As the classical equations for Tajima's D are for single samples sharing a common coalescent history, 
there is more independence in the data than assumed with the classical formula.
This again should make the variance approximation more conservative.

Summing up, the approximate variance in the above equations provides a conservative approximation, 
and the values for Tajima's D will tend to be smaller than those that would be expected 
for an experiment based on individual sequencing of single samples.

Lastly, the PoPoolation code repository contains a plot showing the correlation between the classical Tajima's D 
and the corrected Tajima's D using the equations described above;
please see \href{https://github.com/lczech/popoolation/raw/master/files/correlation_classic_correctedTajimasD.png}{here},
where the x-axis corresponds to the classical value, and the y-axis the the corrected one.
This plot has been made with real-world data from Drosophila with a coverage of 12, 
a window size of 500 and a minimum count of 1.

% % ------------------------------------------------------------------------------------------------------
% %          Simplified Tajima's D
% % ------------------------------------------------------------------------------------------------------

% \subsection{Simplified Tajima's D}
% \label{supp:sec:TajimaD:sub:Simplified}

% \todo{not sure what this is needed for at the moment...}

% ------------------------------------------------------------------------------------------------------
%          PoPoolation Bugs in Tajima's D
% ------------------------------------------------------------------------------------------------------

\subsection{PoPoolation Bugs}
\label{supp:sec:TajimaD:sub:Bugs}

\todo{Rewritten to mention that we are on it, and that we yet have to confirm things, to keep it a bit easier and friendlier. Also if there are really two more bugs as mentioned above, this section becomes a lot longer in the future...}

From our assessment of the \toolname{PoPoolation} code, and from personal communication with Robert Kofler, we suspect that the implementation of the above $\tilde{D}_\text{pool}$ in \toolname{PoPoolation} $\leq$ v1.2.2 contains several bugs, which alter the numerical results of the computation of Tajima's D.
At the moment, we are in contact with Robert Kofler, are still verifying these bugs, and are investigating their consequences.
We here want to thank Robert for his positive reply and his support regarding our questions.

\popoolissue{These are the two bugs that we were already in contact with. We already have stated them in the parts of this document where they come up. We just kept this section here for reference, as it is in our preprint.}

% In the implementation of the above $\tilde{D}_\text{pool}$, 
% there are two bugs in \toolname{PoPoolation} $\leq$ v1.2.2, which alter the numerical results of the computation of Tajima's D.
% Firstly, they compute $\tilde{n}$ not by using the pool size $n$ and the coverage $C$ as stated above, 
% but instead by using the pool size $n$ for both arguments, that is, mistakenly use $C := n$.
% Secondly, instead of computing $\alpha^*$ and $\beta^*$, they only compute $\beta^*$, 
% and use this as the value for $\alpha^*$ as well.
% We have examined the effect of these bugs, and present results of the numerical changes induced by them 
% in \todo{Supplementary document X}.

% \todo{maybe need to explain the two additional bugs here as well that i mentioned above}

%  \todo{@Lucas. Is it necessary to present plots of this? I think we can say this accounts of x percentage of error with coverage or number of individuals below x number}.

% ======================================================================================================
%          F_ST
% ======================================================================================================

\section{Fixation Index \texorpdfstring{\fst}{FST} for Pool-Seq}
\label{supp:sec:FST}

In this section, we will derive unbiased estimators of various measures of heterozygosity in two populations for Pool-sequencing data.
These will then be combined to obtain ``sample-size'' and ``pool-size'' corrected estimators of two definitions of \fst.
On top of these two novel estimators for \fst{} in the pool-sequencing context, we also walk through the two existing estimators as suggested by \citeay{Kofler2011b} and \citeay{Karlsson2007}.
Both are implemented in \toolname{PoPoolation2}, and are called the ``classical'' or ``conventional pool sequencing'' approach, and the ``Karlsson approach adapted to digital data'', respectively, in \citeay{Kofler2011b}.
We compare all four approaches to each other, and show that the ``classical'' approach is biased for lower coverages or small pool sizes, and the Karlsson approach is biased for small pool sizes (bias on the order of 1/pool size).
See also \citeay{Hivert2018} for an assessment of \fst{} in the pool-sequencing context.

\todo{connect to this one as well?! \citeay{Hivert2018} -- I (Lucas) lack the time right now to go through this and work this into the text, but if you want, add it where needed. So far, we just mention it in a general fashion, see above.} 

% ------------------------------------------------------------------------------------------------------
%          Definition of F_ST
% ------------------------------------------------------------------------------------------------------

% \subsection{What the actual \texorpdfstring{F\textsubscript{(ST)}}{F(ST)}?}

\label{supp:sec:FST:sub:Definition}

There are several non-equivalent \emph{definitions} of $\text{F}_\text{ST}$.
The overall goal is to measure some degree of differentiation between two populations, which can be represented as a proportion of variation that cannot be explained by variation within populations.
What is unclear is a proportion of \emph{what} variation?
There are two natural candidates leading to two related, but distinct definitions of $\text{F}_\text{ST}$.
The first definition, which we will call $\text{F}_\text{ST}^\text{Nei}$ following \citeay{Nei1973}, considers the proportion of the total variation in the two populations.
\todo{Jeff, please check the following:}
This statistic is also called $\text{G}_\text{ST}$, see for example Equation (5.5) of \citeay{Hahn2018}.
The second definition, which we will call $\text{F}_\text{ST}^\text{Hudson}$ follwing \citeay{Hudson1992}, considers the proportion of the variation between populations, see also \citeay{Cockerham1969} and \citeay{Weir2002}.
This second definition is also considered in \citeay{Karlsson2007}, which we examine below in \secref{supp:sec:FST:sub:Karlsson}.

To make this more formal, we can consider the probability that two haploids carry different alleles.
We could consider drawing the two haploids from the same population (with the population chosen at random), which we call $\pi_\text{within}$; or we could consider drawing the two haploids from \emph{different} populations, which we call $\pi_\text{between}$; or finally we could consider drawing the two haploids totally at random from either population (potentially the same populations, potentially different populations) which we call $\pi_\text{total}$.
See \citeay{Bhatia2013} for more background information on this.

Our two definitions of $\text{F}_\text{ST}$ are then

\begin{align}
    \text{F}_\text{ST}^\text{Nei}    &= 1 - \frac{\pi_\text{within}}{\pi_\text{total}} \\
    \text{F}_\text{ST}^\text{Hudson} &= 1 - \frac{\pi_\text{within}}{\pi_\text{between}}
\end{align}

If we consider a single locus with up $4$ alleles, with frequencies $f_{\tau{}p}$ (possibly zero) with  $\tau$ denoting the allele with $\tau \in \left\{A, C, G, T\right\}$ and $p$ denoting the population with subscripts $(1)$ and $(2)$, we can calculate the various $\pi$s as follows

\begin{align}
    \label{eq:PiDefs:PiWithin}
    \pi_\text{within} &= \frac{1}{2}\left[\left(1 - \sum_\tau f_{\tau(1)}^2\right) + \left(1 - \sum_\tau f_{\tau(2)}^2\right)\right] \\
    \label{eq:PiDefs:PiBetween}
    \pi_\text{between} &= 1 - \sum_\tau f_{\tau(1)}f_{\tau(2)} \\
    \label{eq:PiDefs:PiTotal}
    \pi_\text{total} &= \frac{1}{2}\pi_\text{within} + \frac{1}{2}\pi_\text{between}
\end{align}

which are then used in our above definitions of \fst.

% ------------------------------------------------------------------------------------------------------
%          Unbiased estimators of the Pi's
% ------------------------------------------------------------------------------------------------------

\subsection{Unbiased estimators of the \texorpdfstring{$\pi$s}{pi's}}
\label{supp:sec:FST:sub:EstimatorsPi}

Since both definitions of $\text{F}_\text{ST}$ rely on these $\pi$s, we will need to derive unbiased estimates for them.
We will show below that the following are unbiased estimators of the corresponding quantities without hats:

\begin{align}
    % \widehat{\pi}_\text{within} &= 
    %     \frac{1}{2} 
    %     \left[ 
    %           \left( \frac{n_{(1)}}{n_{(1)}-1} \right) \left( \frac{C_{(1)}}{C_{(1)}-1} \right) \left( 1 - \sum_{\tau} \left( \frac{c_{\tau(1)}}{C_{(1)}} \right)^2 \right)
    %         + \left( \frac{n_{(2)}}{n_{(2)}-1} \right) \left( \frac{C_{(2)}}{C_{(2)}-1} \right) \left( 1 - \sum_{\tau} \left( \frac{c_{\tau(2)}}{C_{(2)}} \right)^2 \right) 
    %     \right] \\
    \nonumber    
    % Some trickery is needed below for a nice alignment. Not quite perfect, but good enough for now.
    \widehat{\pi}_\text{within} &= 
        \frac{1}{2} 
        \Bigg[                                 \left( \frac{n_{(1)}}{n_{(1)}-1} \right) \left( \frac{C_{(1)}}{C_{(1)}-1} \right) \left( 1 - \sum_{\tau} \left( \frac{c_{\tau(1)}}{C_{(1)}} \right)^2 \right) \Bigg. \\
        \Bigg. &+ \phantom{\frac{1}{2} \Bigg[} \left( \frac{n_{(2)}}{n_{(2)}-1} \right) \left( \frac{C_{(2)}}{C_{(2)}-1} \right) \left( 1 - \sum_{\tau} \left( \frac{c_{\tau(2)}}{C_{(2)}} \right)^2 \right) \Bigg] \\
    \widehat{\pi}_\text{between} &= 
        1 - \sum_{\tau} \left(\frac{C_{\tau,1}}{C_{(1)}}\right)\left(\frac{C_{\tau,2}}{C_{(2)}}\right)  \\
    \widehat{\pi}_\text{total} &= 
        \frac{1}{2}\widehat{\pi}_\text{within} + \frac{1}{2}\widehat{\pi}_\text{between}
\end{align}

In the following, we derive these estimators.

% --------------------------------------------------------
%          Pi Within
% --------------------------------------------------------

\subsubsection*{Unbiased estimator of \texorpdfstring{$\widehat{\pi}_\text{within}$}{Pi Within}}
\label{supp:sec:FST:sub:EstimatorsPi:sub:PiWithin}

We have derived previously that
\[
\mathbb{E}\left[\left(\frac{n_{(1)}}{n_{(1)}-1}\right)\left(\frac{C_{(1)}}{C_{(1)}-1}\right)\left(1 - \sum_{\tau}\left(\frac{C_{\tau,1}}{C_{(1)}}\right)^2\right) \right] = \left(1 - \sum_\tau f_{\tau, 1}^2\right),
\]
within a single population.  It follows immediately that averaging these estimators across the two populations is unbiased for $\pi_\text{within}$.

% --------------------------------------------------------
%          Pi Between
% --------------------------------------------------------

\subsubsection*{Unbiased estimator of \texorpdfstring{$\widehat{\pi}_\text{between}$}{Pi Between}}
\label{supp:sec:FST:sub:EstimatorsPi:sub:PiBetween}

\todo{I think some of the capital C's here and in surrounding sections should be lower case.}

Since the two pools are independent, we have that
\[
\mathbb{E}\left[\widehat{\pi}_\text{between}\right] = 1- \sum_\tau \mathbb{E}\left[\left(\frac{C_{\tau,1}}{C_{(1)}}\right)\right]\mathbb{E}\left[\left(\frac{C_{\tau,2}}{C_{(2)}}\right)\right].
\]
The frequency of alleles within a pool is an unbiased estimate for the frequency in the population, so
\[
\mathbb{E}\left[\left(\frac{C_{\tau,p}}{C_{(p)}}\right)\right] = f_{\tau, p},
\]
showing that $\widehat{\pi}_\text{between}$ is unbiased for $\pi_\text{between}$.

% --------------------------------------------------------
%          Pi Total
% --------------------------------------------------------

\subsubsection*{Unbiased estimator of \texorpdfstring{$\widehat{\pi}_\text{total}$}{Pi Total}}
\label{supp:sec:FST:sub:EstimatorsPi:sub:PiTotal}

That $\widehat{\pi}_\text{total}$ is unbiased for $\pi_\text{total}$ follows immediately from the definition of $\pi_\text{total}$ in \eqnref{eq:PiDefs:PiTotal} and the unbiasedness of $\widehat{\pi}_\text{within}$ and $\widehat{\pi}_\text{between}$.

% ------------------------------------------------------------------------------------------------------
%          Estimating FST
% ------------------------------------------------------------------------------------------------------

\subsection{Final unbiased estimators of \texorpdfstring{$\text{F}_\text{ST}$}{FST} per SNP and per window}
\label{supp:sec:FST:sub:EstimatorFST}

These estimators then immediately suggest the following ratio estimators for the different definitions of $\text{F}_\text{ST}$:

\begin{align}
    \widehat{\text{F}}_\text{ST}^\text{Nei} &= 1 - \frac{\widehat{\pi}_\text{within}}{\widehat{\pi}_\text{total}}\\
    \widehat{\text{F}}_\text{ST}^\text{Hudson} &= 1 - \frac{\widehat{\pi}_\text{within}}{\widehat{\pi}_\text{between}}
\end{align}

All of this has been for a single site, but we are often interested in combining information across SNPs within a window $W$ (or possibly genome wide).
In such a case, define $\widehat{\pi}^\ell_\text{within}$ to be $\widehat{\pi}_\text{within}$ as above but for SNP $\ell \in W$.
Define $\widehat{\pi}^\ell_\text{between}$ and $\widehat{\pi}^\ell_\text{total}$ analogously.
We then combine information across the SNPs in the window $W$ as

\begin{align}
    \widehat{\text{F}}_\text{ST}^\text{Nei}    &= 1 - \frac{\sum_{\ell \in W} \widehat{\pi}^\ell_\text{within}} {\sum_{\ell \in W} \widehat{\pi}^\ell_\text{total}} \\
    \widehat{\text{F}}_\text{ST}^\text{Hudson} &= 1 - \frac{\sum_{\ell \in W} \widehat{\pi}^\ell_\text{within}} {\sum_{\ell \in W} \widehat{\pi}^\ell_\text{between}}
\end{align}

See \citeay{Bhatia2013} for a practical and theoretical justification for using this ``ratio of averages'' instead of using an ``average of ratios''.
These are our asymptotically unbiased estimators for \fst{} for Pool-seq data, which take the finite sampling of individuals from the population, and the finite sampling of reads from each individual in the pool, into account.

\todo{Need to update later:}
At the time of writing, we have only theoretically derived these estimators, but have not yet implemented them in our software.
\todo{According to preliminary simulations, these are not biased... etc}

% ------------------------------------------------------------------------------------------------------
%          PoPoolation2 Estimator of FST
% ------------------------------------------------------------------------------------------------------

\subsection{Estimator of \texorpdfstring{$\text{F}_\text{ST}$}{FST} as implemented in PoPoolation2}
\label{supp:sec:FST:sub:PoPoolation2Estimator}

The implementation in \toolname{PoPoolation2} \cite{Kofler2011b} offers two ways to estimate \fst{}:
What they call the ``classical'' or ``conventional'' approach by \citeay{Hartl2007}, and an approach adapted to digital data following \citeay{Karlsson2007}.
In this and the next section, we discuss these estimators.
\todo{contingent on Jeff's simulations, i.e., to be added later: We also show that they are biased, and hence recommend to use our novel estimates as introduced above instead.}
For comparability and historical backwards compatibility, we however still offer both these estimators in our implementation.

\todo{see if we actually get to do this in time:}
Furthermore, the PoPoolation equations document explains derivations of equations for Pool-seq corrected estimators of \fst, which however to the best of our knowledge are not actually implemented in either \toolname{PoPoolation} nor \toolname{PoPoolation2}.
We here still walk through these, see \secref{supp:sec:PoPoolation2Equations}.

First, we present the ``classical'' approach as implemented in \toolname{PoPoolation2}, labelled with superscript ``PoPool'' here.
We compute \fst{} for two subpopulations, which we here again denote with subscripts $(1)$ and $(2)$, and the total population with $(T)$. We expect poolsizes $n >= 2$.

For each SNP in a given window, \toolname{PoPoolation2} computes:
% \verb|f_st_conventional_pool_pi_snp|

\begin{align}
    \label{eq:PoPoolation2FstPi:1}
    \widehat{\pi}_{(1)}^\text{PoPool}  &= \frac{C_{(1)}}{C_{(1)}-1} \cdot  \left( 1 - \sum_\tau f_{\tau(1)}^2 \right) \\
    \label{eq:PoPoolation2FstPi:2}
    \widehat{\pi}_{(2)}^\text{PoPool}  &= \frac{C_{(2)}}{C_{(2)}-1} \cdot  \left( 1 - \sum_\tau f_{\tau(2)}^2 \right) \\
    \label{eq:PoPoolation2FstPi:T}
    \widehat{\pi}_{(T)}^\text{PoPool}  &= \frac{C_{(T)}}{C_{(T)}-1} \cdot  \left( 1 - \sum_\tau f_{\tau(T)}^2 \right) \\
    \nonumber
    \mbox{with} \\
    \nonumber
    C_{(T)} &= \mbox{min} \left( C_{(1)}, C_{(2)} \right) \\
    \nonumber
    f_{\tau(T)} &= \frac{1}{2} \cdot \left( f_{\tau(1)} + f_{\tau(2)} \right)
\end{align}

These quantities are accumulated over the window $W$:
% \verb|f_st_conventional_pool|

\begin{align}
    \label{eq:PoPoolation2FstWindow:1}
    \widehat{\pi}_{W(1)}^\text{PoPool} &= \frac{n_{(1)}}{n_{(1)}-1} \cdot \sum_{W} \pi_{(1)} \\
    \label{eq:PoPoolation2FstWindow:2}
    \widehat{\pi}_{W(2)}^\text{PoPool} &= \frac{n_{(2)}}{n_{(2)}-1} \cdot \sum_{W} \pi_{(2)} \\
    \label{eq:PoPoolation2FstWindow:T}
    \widehat{\pi}_{W(T)}^\text{PoPool} &= \frac{n_{(T)}}{n_{(T)}-1} \cdot \sum_{W} \pi_{(T)} \\
    \nonumber
    \mbox{with} \\
    \nonumber
    n_{(T)} &= \mbox{min} \left( n_{(1)}, n_{(2)} \right)
\end{align}

Finally, the estimate of \fst is computed as:

\begin{align}
    \label{eq:PoPoolation2FstEst}
    \widehat{\text{F}}_\text{FST}^\text{PoPool} &= \frac{\pi_{W(T)} - \frac{1}{2} \left( \pi_{W(1)} + \pi_{W(2)} \right)}{\pi_{W(T)}}
\end{align}

\todo{Jeff, please check:} Because this estimator uses the minimum coverage and minimum pool size for either of the populations to calculate $\pi_{W(T)}$, it produces biased $F_{ST}$ values for small pool sizes or coverages. This was also pointed out by \citeay{Hivert2018}. It is therefore recommended to use the unbiased estimator presented above.
\todo{(Moi) I have given it a try to explain the bias, I think is just the minimum coverage or n simplification.}
\todo{Jeff, please check and expand the rational of the last sentence as needed.}
\todo{TBD: See \secref{supp:sec:FST:sub:Comparison} for an evaluation of the biases introduced by this estimator.}

% Moi's rendering of the equations. Keeping it here for reference, in case it's later needed.

% We create an unbiased estimator based on $\text{F}_\text{ST}$'s definition based on nucleotide diversity within a sub-population $S$ and across a total set of populations $T$ (see Equation (5.5) of \citeay{Hahn2018}):
% \begin{align}
%     \text{F}_\text{ST} &= \frac{\theta_{\pi,T} - \theta_{\pi,S}}{\theta_{\pi,T}}
% \end{align}
% Since we have already derived unbiased estimators for $\theta_\pi$, given two libraries of DNA sequences from two Pool-seq populations with their corresponding $u$ alternative alleles, $v$ reference alleles, $C$ coverages, and $n$ individuals, we can calculate: 
% \begin{align}
%   \text{F}_\text{ST} &= 
%     \frac{
%         \theta_\pi(u_1+u_2,v_1+v_2,C_1+C_2,n_1+n_2) - 
%         \frac{1}{2}(
%             \theta_\pi(u_1,v_1,C_1,n_1) + 
%             \theta_\pi(u_2,v_2,C_2,n_2)
%         )
%     }{
%      \theta_\pi(u_1+u_2,v_1+v_2,C_1+C_2,n_1+n_2)
%     }
% \end{align}

% ------------------------------------------------------------------------------------------------------
%          FST Asymptotically Unbiased, Karlsson
% ------------------------------------------------------------------------------------------------------

\subsection{Asymptotically Unbiased Estimator of \texorpdfstring{\fst}{FST} by Karlsson \textit{et al.}}
\label{supp:sec:FST:sub:Karlsson}

Another estimator for \fst{} that is offered in \toolname{PoPoolation2} is based on the equations used in \citeay{Karlsson2007}, see the last page of the Supplemental Information of Karlsson \textit{et al.} for their derivation.
We here briefly also go through the derivation.
% We also offer this estimator in our implementation, and briefly derive it here.

We here call this estimator using the superscript ``Karlsson'', which is again defined for two subpopulations denoted with subscripts $(1)$ and $(2)$.
% We expect poolsizes $n >= 2$.
We are here only looking at biallelic SNPs.
Instead of $\tau$ for the four nucleotides, we hence use $u$ for the major and $v$ for the minor allele again, where $u$ is the allele with the higher average frequency in the two subpopulations (as opposed to the allele with the highest total count).

We start with the definition of $\text{F}_\text{FST}^\text{Karlsson}$ from Karlsson \textit{et al.} for the SNPs in a window $W$:

\begin{align}
    \label{eq:FstK}
    \text{F}_\text{FST}^\text{Karlsson} &= \frac{\sum_W N_k}{\sum_W D_k}
\end{align}

where the the numerator $N_k$ and denominator $D_k$ for a single site $k$ in $W$ are:

\begin{align}
    \label{eq:FstNk}
    N_k &= v_{(1)} \cdot ( u_{(2)} - u_{(1)} ) ~+~ v_{(2)} \cdot ( u_{(1)} - u_{(2)} ) \\
    \label{eq:FstDk}
    \nonumber
    D_k &= v_{(1)} u_{(2)} + u_{(1)} v_{(2)} \\
        &= N_k + v_{(1)} u_{(1)} + v_{(2)} + u_{(2)}
\end{align}

These are estimated as follows, using the numerator $\hat{N}_k$ and denominator $\hat{D}_k$ at a single site:

\begin{align}
    \label{eq:FstKnh}
    \hat{N}_k &= \left( \frac{u_{(1)}}{C_{(1)}} - \frac{u_{(2)}}{C_{(2)}} \right)^2 - \left( \frac{h_{(1)}}{C_{(1)}} + \frac{h_{(2)}}{C_{(2)}} \right) \\
    \label{eq:FstKdh}
    \hat{D}_k &= \hat{N}_k + h_{(1)} + h_{(2)}
    \intertext{with two additional helpers:}
    \nonumber
    h_{(1)} &= \frac{u_{(1)} \cdot v_{(1)}}{ C_{(1)} \cdot \left( C_{(1)} -1 \right)} \\
    \nonumber
    h_{(2)} &= \frac{u_{(2)} \cdot v_{(2)}}{ C_{(2)} \cdot \left( C_{(2)} -1 \right)}
\end{align}

And finally, these are used to compute the asymptotically unbiased estimator $\widehat{\text{F}}_\text{FST}^\text{Karlsson}$ for a window $W$:

\begin{align}
    \label{eq:FstEstK}
    \widehat{\text{F}}_\text{FST}^\text{Karlsson} &= \frac{\sum_W \hat{N}_k}{\sum_W \hat{D}_k}
\end{align}

According to Karlsson \textit{et al.}, when the coverages $C_{(1)}$ and $C_{(2)}$ (called ``sample sizes'' there) are equal, the estimator reduces to the estimator of \fst{} given by \citeay{Weir2002}.
Karlsson \textit{et al.} further state that by the Lehmann-Scheff\'{e} theorem \cite[Theorem 4.2.2]{Bickel1977}, it follows that $\hat{N}_k$ and $\hat{D}_k$ are uniformly minimum variance unbiased estimators of $N_k$ and $D_k$, respectively, and hence conclude that their estimator \^{F}\textsubscript{ST,K} is also asymptotically unbiased.

\todo{Jeff, please check the following:}
The estimator above hence follows what we called the $\text{F}_\text{ST}^\text{Hudson}$ definition.
It however assumes the pool size to be infinite, that is, it is missing Bessel's correction for pool size.
Apart from that, it is identical to our estimator $\widehat{\text{F}}_\text{ST}^\text{Hudson}$ as explained in \secref{supp:sec:FST:sub:EstimatorFST}.

\todo{Reactivate the below Comparison paragraph if we have simulations to show}

% ------------------------------------------------------------------------------------------------------
%          Comparison of the Estimators and their Biases
% ------------------------------------------------------------------------------------------------------

% \subsection{Comparison of the Estimators and their Biases}
% \label{supp:sec:FST:sub:Comparison}

% As mentioned above, both the PoPoolation2 estimator and the Karlsson estimator have biases.
% We here explore their effects via simulations, and further show that our estimators are unbiased under the Pool-seq assumptions.

% \todo{todo todo to do to do to do to do to doooooo}

% % We hence highly recommend to use our novel estimators over the ones implemented in \toolname{PoPoolation2}.

% ------------------------------------------------------------------------------------------------------
%          PoPoolation2 Equations Document
% ------------------------------------------------------------------------------------------------------
\todo{(Moi) I think is fine to just say this and not go through equations ;) }

\section{PoPoolation2 Equations Document}
\label{supp:sec:PoPoolation2Equations}

The PoPoolations equation document also presents some simplifications and related equations that to the best of our knowledge are not implemented in their software.
We hence do not go through them in detail here, but still want to mention them, in case they might be useful for others.

\begin{itemize}
  \item They present simplified versions of $\theta_\pi$, $\theta_w$, and Tajima's D, which assume that allele frequency distribution in the reads is about the same as in the real population, and hence arrives at a simpler computation at the cost of some error. These are also useful for individual sequencing.
  \item As mentioned above in \secref{supp:sec:TajimaD:sub:PoolSequencingCorrection}, the document presents an approach to computing Tajima's D based on its variance, and extends this to windows, but (to the best of our knowledge) does not implement this, and instead implement their approach based on \citeay{Achaz2008}.
  \item They present an approach for computing \fst{} for $J$ pool-sequenced populations (instead of just two as presented above), extend this approach to large regions as well as single SNPs, and introduce weights that take the number of sequenced individuals in each population into account. More work is needed to compare this approach to their implementation and to our novel estimators.
\end{itemize}

These alternative approaches however need further assessment and comparison to the other approaches presented here.

\popoolissue{Lastly, we wanted to note that to us it seems that some equations in the PoPoolation equations document are not actually implemented in the code, and that the code contains compuations that are not in the document. We hence think that those were never intended to go hand in hand, and that hence the equations document is also not part of the official publication, but merely found in the code repository. Is that assessment correct, or are there other reasons for the divergence between the two?}

% The Pool-seq corrected equations for $\text{F}_\text{ST}$ available in the pdf document of PoPoolation's code repository are not implemented, to the best of our knowledge, in \toolname{PoPoolation} nor in \toolname{PoPoolation2} software. 
% Here we describe the methods implemented in \toolname{PoPoolation2} code, explain the rationale, and expand on calculation options of our own implementation.
% See \citeay{Bhatia2013} for a good introduction to the topic and the confusion around \fst.

% The below section names are what the PoPoolation equations document offers.
% At the moment however, it seems that the implementation (which is not in \toolname{PoPoolation}, but in \toolname{PoPoolation2})
% differs from the equations that they have in their document. 
% So, instead of copying the above equations from their document to here for somthing that is not even implemented, 
% as of now, we leave the above blank.

% Instead, we here write down the equations that are actually implemented in \toolname{PoPoolation2}, based on the code.

% % --------------------------------------------------------
% %          Large Regions
% % --------------------------------------------------------

% \subsubsection{Large Regions}
% \label{supp:sec:FST:sub:LargeRegions}

% % --------------------------------------------------------
% %          Simplified
% % --------------------------------------------------------

% \subsubsection{Simplified Regional \texorpdfstring{\fst}{FST}}
% \label{supp:sec:FST:sub:Simplified}

% % (also good for individual sequencing again)

% % --------------------------------------------------------
% %          Single SNPs
% % --------------------------------------------------------

% \subsubsection{Single SNPs}
% \label{supp:sec:FST:sub:SingleSNPs}

% % --------------------------------------------------------
% %          Weights
% % --------------------------------------------------------

% \subsubsection{Weights}
% \label{supp:sec:FST:sub:Weights}

% % ======================================================================================================
% %          SCRATCH
% % ======================================================================================================

\todo{the below are (commented out) reverse-engineered equations of the non-pool-seq-corrected equations that are also implemented in popoolation. we probably do not need them, but i want to keep them here for reference, if needed.}

% \section{SCRATCH}
% \label{supp:sec:SCRATCH}

% \todo{scratch space added by Lucas to write down PoPoolation equations as implemented}

% \subsection{Classical}
% \label{supp:sec:SCRATCH:sub:Classical}

% these are what popoolation calls ``classical'' computations, that is, without pool seq correction.
% they still use some equations that do not seem 100\% typical to me, but let's see.

% \textbf{classical pi:}

% \begin{align}
%     \theta_{\pi,Classical} &=  1 - \frac{2}{C(C-1)} \sum_\tau \frac{c_\tau (c_\tau - 1)}{2}
% \end{align}

% the call this computation ``average pairwise difference''.
% the 2s obviously cancel out, but i kept them here, as this is what's implemented.


% \textbf{classical theta (watterson): }

% \begin{align}
%     \theta_{W,Classical} &=  \frac{\text{\#SNPs}}{a_1(n_m)}
% \end{align}

% where $n_m$ is the median of all coverages in the given window,
% and $a_1$ again the harmonic as defined above.
% no idea if this is a thing that one does... but that's what they do.


% \textbf{classical D:}

% \begin{align}
%     D &=  \frac{\theta_{\pi,Classical} - \theta_{W,Classical}}{V}
% \end{align}

% where $V$ is called ``sqrt variance'', probably meaning square root of the variance, i.e., standard deviation?! and computed as:

% \begin{align}
%     V   &= \sqrt{ e_1 \cdot \text{\#SNPs} + e_2 \cdot \text{\#SNPs} \cdot (\text{\#SNPs} -1) } \\
%     e_1 &= \frac{ \frac{n_m + 1}{3 \cdot (n_m - 1)} - \frac{1}{a_1(n_m)} }{a_1(n_m)} \\
%     e_2 &= \frac{ \frac{2(n_m^2 + n_m + 3)}{9n_m(n_m-1)} - \frac{n_m+2}{a_1(n_m) \cdot n_m} + \frac{a_2(n_m)}{a_1(n_m)^2} }{ a_1(n_m)^2 + a_2(n_m) }
% \end{align}

% again using the median coverage $n_m$, and (squared) harmonic $a_1$ and $a_2$.


% ======================================================================================================
%          Conclusion
% ======================================================================================================

% \section{Conclusion}
% \label{supp:sec:Conclusion}

% \todo{needs updating later}

% We have here presented our current understanding of the equations and the implementation of PoPoolation \cite{Kofler2011a,Kofler2011b} for population genetic measures $\theta_\pi$, $\theta_w$, Tajima's D, and \fst{}. We have further introduced two novel estimators of \fst{} for pool-sequenced data.

% We invite everyone to join the discussion and to help establishing well-founded theory of these measures for pool sequencing.
% Hence, this document is a draft, and more work is needed to thoroughly assess the whole situation.
% % There might be bugs in the \toolname{PoPoolation} implementation of Tajima's D. 


% ######################################################################################################################
%         Appendices
% ######################################################################################################################

% \clearpage

% Bibliography
% \bibliographystyle{pnas-new}
\bibliographystyle{natbib}
% \bibliographystyle{apalike}
% \bibliographystyle{myabbrvnat}

\bibliography{references}

\end{document}
